### ISSUE TITLE
Installation issue: OpenMPI has implicit runtime dependency on ssh

### ISSUE THREAD

**Issue by sethrj** [LLNL/spack#18084]
I'm using Spack to install OpenMPI on a Docker container (nvidia/cuda:11.0-devel-ubuntu18.04), but that image didn't by default contain ssh, leading to a runtime error (similar to https://github.com/open-mpi/ompi/issues/3625):
```
--------------------------------------------------------------------------
The value of the MCA parameter "plm_rsh_agent" was set to a path
that could not be found:

  plm_rsh_agent: ssh : rsh

Please either unset the parameter, or check that the path is correct
--------------------------------------------------------------------------
[dc96570fc736:39976] [[INVALID],INVALID] FORCE-TERMINATE AT Not found:-13 - error plm_rsh_component.c(327)
[dc96570fc736:39976] *** Process received signal ***
[dc96570fc736:39976] Signal: Segmentation fault (11)
[dc96570fc736:39976] Signal code: Address not mapped (1)
[dc96570fc736:39976] Failing at address: (nil)
[dc96570fc736:39976] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x128a0)[0x7f8a0b9048a0]
[dc96570fc736:39976] *** End of error message ***
[dc96570fc736:39975] [[INVALID],INVALID] ORTE_ERROR_LOG: Unable to start a daemon on the local node in file ess_singleton_module.c at line 532
[dc96570fc736:39975] [[INVALID],INVALID] ORTE_ERROR_LOG: Unable to start a daemon on the local node in file ess_singleton_module.c at line 166
--------------------------------------------------------------------------
It looks like orte_init failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during orte_init; some of which are due to configuration or
environment problems.  This failure appears to be an internal failure;
here's some additional information (which may only be relevant to an
Open MPI developer):

  orte_ess_init failed
  --> Returned value Unable to start a daemon on the local node (-127) instead of ORTE_SUCCESS
--------------------------------------------------------------------------
--------------------------------------------------------------------------
It looks like MPI_INIT failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during MPI_INIT; some of which are due to configuration or environment
problems.  This failure appears to be an internal failure; here's some
additional information (which may only be relevant to an Open MPI
developer):

  ompi_mpi_init: ompi_rte_init failed
  --> Returned "Unable to start a daemon on the local node" (-127) instead of "Success" (0)
--------------------------------------------------------------------------
*** An error occurred in MPI_Init
*** on a NULL communicator
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)
```

Installing ssh via apt-get (`LD_LIBRARY_PATH= apt-get update;LD_LIBRARY_PATH= apt-get -yqq install --no-install-recommends ssh`) *or* adding `openssh` to my spack environment fixes the error.

Do we need to add a `run` dependency on openmpi with the `auto` fabric to include openssh? Is that a silly requirement because most systems have openssh already installed?

### Steps to reproduce the issue

```console
$ spack install openmpi
```

### Information on your system

* **Spack:** 0.15.4
* **Python:** 3.6.9
* **Platform:** linux-ubuntu18.04-haswell

### Additional information

@hppritcha

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [ ] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by sjpb** [comment]
From [here](https://www.open-mpi.org/faq/?category=rsh#rsh-not-ssh) the `plm_rsh_agent: ssh : rsh` means openmpi should be falling back to rsh if it can't find ssh - this is the default behavior. So presumably you haven't got rsh either?

However that does show ssh isn't *actually* a requirement. Also if running openmpi under slurm then the [recommended methods of job launch for modern openmpi](https://slurm.schedmd.com/mpi_guide.html) don't actually need ssh either. I agree an install which doesn't result in a working program is problematic though!

**Comment by sethrj** [comment]
@sjpb Good point about `rsh`. No, this is based on the nvidia docker image, using spack to build an environment.

**Comment by haampie** [comment]
I ran into the same problem. What is necessary to fix this? Is `depends_on('openssh', type='run')` too heavy?

**Comment by sethrj** [comment]
@haampie maybe a runtime dependency for ssh as a new variant?

**Comment by sethrj** [comment]
Fixed by #22115 .
